### PR TITLE
Fix stats-in-tracer tests for the .NET library

### PR DESCRIPTION
**Pull request by zacharycmontoya**
## Description

Makes the required changes for the dotnet tests to pass all stats-in-tracer tests (except for the relative error test case that is skipped for all tracers)

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [X] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

**Comment by zacharycmontoya** [comment]
If the tests are passing, can someone merge this? I will be on PTO until the new year

**Review by cbeauchesne — APPROVED** [review]

**Comment by cbeauchesne** [comment]
ok for me, waiting for @Kyle-Verhoog approval

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**
